memo.rb

```diff
@@ -53,29 +63,22 @@ def write_memos(edited_memos)
 
 get '/editing/*' do |id|
   @title = '編集'
-  @memo = get_memo(id)
+  @memo = select_one(id)
   erb @memo ? :editing : :notFound
 end
 
 post '/memo' do
-  id = SecureRandom.uuid
-  edited_memos = read_memos
-  edited_memos << { 'id' => id, 'title' => params[:title], 'text' => params[:text] }
-  write_memos(edited_memos)
-  redirect "/memo/#{id}"
+  returning = insert_memo(params.slice(:title, :main_text))
+  redirect "/memo/#{returning.first['id']}"
 end
 
 patch '/memo/*' do |id|
-  edited_memos = read_memos.map do |memo|
-    memo['id'] == id ? { 'id' => id, 'title' => params[:title], 'text' => params[:text] } : memo
-  end
-  write_memos(edited_memos)
+  update_memo(params.slice(:title, :main_text).merge({ id: }))
```

Comment:
```suggestion
  update_memo(params.slice(:title, :main_text).merge(id:))
```

引数の最後にハッシュが来るときは中括弧を省略できます（チェリー本参照）